designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+# net5.0 TFM Platform and PlatformVersion implementation
+
+This document is meant to describe the various parts of the code that need to be updated in order to support platforms in net5.0 target framework identifiers. That is, the optional part after a `-` following `net5.0`. For example, `net5.0-ios14.1`. Note that this ONLY covers the platform part, because [initial support for `net5.0` has already been added](https://github.com/NuGet/NuGet.Client/pull/3265).
+
+For more details on the feature itself, please refer to [the full spec](https://github.com/terrajobst/designs/blob/master/accepted/2020/net5/net5.md).
+
+## class NuGetFramework
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs
+
+### Fields
+
+Three additional fields will be added:
+
+```csharp
+        /// <summary>
+        /// Framework Platform (net5.0+)
+        /// </summary>
+        public string Platform
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Framework Platform Version (net5.0+)
+        /// </summary>
+        public Version PlatformVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// True if platform is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatform
+        {
+            get { return !string.IsNullOrEmpty(Platform); }
+        }
+
+        /// <summary>
+        /// True if platform version is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatformVersion
+        {
+            get { return IsNet5Era && !(PlatformVersion.Major == 0 && PlatformVersion.Minor == 0); }
+        }
+```
+
+### Constructors
+
+An additional constructor should be added that accepts `Version platformVersion` as its 4th argument. The current code in `NuGetFramework/3` should be moved to this new constructor, and `string profile` should be renamed to `string profileOrPlatform`. This field will be EITHER a legacy profile or, if net5.0+, a platform identifier.

```

Comment:
I think platform identifier and platform version should be two new parameters, rather than re-using profile depending on context (framework & version). With the current implementation, it makes it impossible for non-.NET TFMs (like `tizan` or `unity`) to use platform identifier and platform version, if they would ever want to in the future. By having profile and platform as separate parameters, by convention .NET will only use one of the two, depending on the framework version, but it doesn't prohibit other use-cases.

Reply:
I think they should be new parameters as well.


I think we should just have a 4 val constructor, TFI, TFV, TPI & TPV.